account/accounts.go

```diff
@@ -55,15 +51,29 @@ type Manager struct {
 }
 
 // NewManager returns new node account manager.
-func NewManager(geth GethServiceProvider) *Manager {
-	manager := &Manager{
-		geth: geth,
-	}
-
-	accountsGenerator := generator.New(manager)
-	manager.accountsGenerator = accountsGenerator
+func NewManager() *Manager {
+	m := &Manager{}
```

Comment:
```suggestion
	m := &Manager{
		accountsGenerator: generator.New(m),
	}
```

Reply:
there is a circular reference, generator.New takes pointer to Manager as an input